core/execution/liquidation/engine.go

```diff
@@ -132,14 +128,21 @@ func (e *Engine) Update(cfg *types.LiquidationStrategy) {
 	e.cfg = cfg
 }
 
-func (e *Engine) OnTick(ctx context.Context, now time.Time) (*types.Order, error) {
-	if e.pos.open == 0 || e.as.InAuction() || e.nextStep.After(now) {
+func (e *Engine) OnTick(ctx context.Context, now time.Time, midPrice *num.Uint) (*types.Order, error) {
+	if e.pos.open == 0 || e.as.InAuction() || e.nextStep.After(now) || midPrice.IsZero() {
 		return nil, nil
 	}
-	minP, maxP, err := e.ml.ValidOrdersPriceRange()
-	if err != nil {
-		return nil, err
+
+	one := num.DecimalOne()
+	// get the min/max price from the range based on slippage parameter
+	mpDec := num.DecimalFromUint(midPrice)
+	minP := num.UintZero()
+	if e.cfg.DisposalSlippage.LessThan(one) {
+		minD := mpDec.Mul(one.Sub(e.cfg.DisposalSlippage))
+		minP, _ = num.UintFromDecimal(minD)
```

Comment:
I still don't understand why we have the if when `DisposalSlippage` is _always_ bigger then zero. Shouldn't it just be the below?

```suggestion
minD := num.MaxD(num.DecimalZero(), mpDec.Mul(one.Sub(e.cfg.DisposalSlippage)))
minP, _ = num.UintFromDecimal(minD)
```

Reply:
because if disposal slippage > 1, the min price calculated here will _always_ be negative, and there's no point in calculating it, just set the min price to 0, and be done with it.